submarine-server/server-submitter/submitter-yarnservice/src/main/java/org/apache/submarine/server/submitter/yarnservice/YarnServiceUtils.java

```diff
@@ -30,35 +30,31 @@
  * based on the provided parameters.
  */
 public final class YarnServiceUtils {
-  private YarnServiceUtils() {
-  }
-
-  // This will be true only in UT.
-  private static AppAdminClient stubServiceClient = null;
-
-  static AppAdminClient createServiceClient(
-      Configuration yarnConfiguration) {
-    if (stubServiceClient != null) {
-      return stubServiceClient;
-    }
+  private static AppAdminClientFactory appAdminClientFactory =
+          new AppAdminClientFactory();
 
-    return AppAdminClient.createAppAdminClient(DEFAULT_TYPE, yarnConfiguration);
+  private YarnServiceUtils() {
   }
 
   @VisibleForTesting
-  public static void setStubServiceClient(AppAdminClient stubServiceClient) {
-    YarnServiceUtils.stubServiceClient = stubServiceClient;
+  public static void setAppAdminClientFactory(
+          AppAdminClientFactory appAdminClientFactory) {
+    YarnServiceUtils.appAdminClientFactory = appAdminClientFactory;
   }
 
-  public static String getDNSName(String serviceName,
-      String componentInstanceName, String userName, String domain, int port) {
-    return componentInstanceName + getDNSNameCommonSuffix(serviceName, userName,
-        domain, port);
+  static AppAdminClient createServiceClient(
+          Configuration yarnConfiguration) {
+    return appAdminClientFactory.createDefault(yarnConfiguration);
   }
 
   public static String getDNSNameCommonSuffix(String serviceName,
-      String userName, String domain, int port) {
+                                              String userName, String domain, int port) {
```

Comment:
should remove extra whitespace

Reply:
Done (and also removed extra whitespaces from the whole class).